rest/src/main/java/discord4j/rest/request/RequestStream.java

```diff
@@ -132,7 +132,14 @@ void push(RequestCorrelation<T> request) {
     }
 
     void start() {
-        requestQueue.requests().subscribe(new RequestSubscriber(rateLimitStrategy));
+        requestQueue.requests()
+                .doOnDiscard(RequestCorrelation.class, this::onDiscard)
+                .subscribe(new RequestSubscriber(rateLimitStrategy));
+    }
+
+    private void onDiscard(RequestCorrelation<?> requestCorrelation) {
+        requestCorrelation.getResponse().onComplete();
```

Comment:
While this works well, honestly I'm tempted to make it error with an exception instead of .onComplete() it. That way we can make difference between a request that has been discarded and a request that completed empty because of `ResponseFunction.emptyIfNotFound()` for example. But I don't know what type of exception I could put. Any suggestions?

Reply:
I agree. Could be `DiscardedRequestException`.. something along those lines.

Reply:
Changed